gemini/use-cases/retrieval-augmented-generation/utils/intro_multimodal_rag_utils.py

```diff
@@ -400,17 +386,29 @@ def get_gemini_response(
         The generated text as a string.
     """
 
-    generation_config = {"max_output_tokens": 2048, "temperature": 0.1}
+    # generation_config = {"max_output_tokens": 2048, "temperature": 0.1}
+    # print(generation_config)
+    # print(safety_settings)
 
     if stream:
         response = generative_multimodal_model.generate_content(
             model_input,
             generation_config=generation_config,
             stream=stream,
+            safety_settings=safety_settings,
         )
         response_list = []
+
         for chunk in response:
-            response_list.append(chunk.text)
+            try:
+                response_list.append(chunk.text)
+            except Exception as e:
+                print(
+                    "Exception occurred while calling gemini. Something is wrong. Lower the safety thresholds [safety_settings: BLOCK_NONE ] if not already done. -----",
+                    e,
+                )
+                response_list.append("Exception occurred")
+                continue
         response = "".join(response_list)
     else:
         response = generative_multimodal_model.generate_content(
```

Comment:
Should safety settings be sent in this call as well?

Also, you should be able to move this function call outside of the if-else and pass stream as a parameter.

Reply:
Fixed!